app/lib/package/backend.dart

```diff
@@ -56,6 +56,8 @@ final _defaultMaxVersionsPerPackage = 1000;
 final Logger _logger = Logger('pub.cloud_repository');
 final _validGithubUserOrRepoRegExp =
     RegExp(r'^[a-z0-9\-\._]+$', caseSensitive: false);
+final _validGithubVersionPattern =
+    RegExp(r'^[a-z0-9\-\._]+$', caseSensitive: false);
```

Comment:
Tiny nit, I don't think you have to escape the dot:
```suggestion
    RegExp(r'^[a-z0-9\-._]+$', caseSensitive: false);
```